src/main/java/spoon/SpoonException.java

```diff
@@ -21,4 +28,19 @@ public SpoonException(Throwable e) {
 	public SpoonException(String msg, Throwable e) {
 		super(msg, e);
 	}
+	/**
+	 * Handles a JLSViolation according to the environment settings. If {@link Environment#getIgnoreSyntaxErrors()} is set to true, the exception is ignored.
+	 * Otherwise, the exception is thrown.
+	 * @param holder  an object with access to factory and environment.
+	 * @param reason  the reason for the exception.
+	 */
+	public static void handleJLSViolation(FactoryAccessor holder, String reason) {
```

Comment:
I don't think this method belongs here either. This makes `SpoonException` dependent on its subtype, which is undesirable and IMO violates SRO (`SpoonException` might have to change if `JLSViolation` changes).

If this method should exist, and should be the prime way of throwing a `JLSViolation`, I would put it as a static method on `JLSViolation` and make its constructor private. I.e. like so:

```java
public class JLSViolation extends SpoonException {

	private JLSViolation(String msg) {
		super(msg);
	}

	public static void throwIfSyntaxErrorsAreNotIgnored(FactoryAccessor holder, String reason) {
		// logic    
	}

}
```

Then usage becomes a bit easier to read, I think:

```java
// current
SpoonException.handleJLSViolation(...);

// my proposition
JLSViolation.throwIfSyntaxErrorsAreNotIgnored(...);
```

What do you think? Agree, disagree?

Reply:
100% correct, I moved the method and changed the name to your better name.